src/main/java/de/tum/cit/ase/ares/api/architecture/java/archunit/postcompile/TransitivelyAccessesMethodsCondition.java

```diff
@@ -0,0 +1,180 @@
+package de.tum.cit.ase.ares.api.architecture.java.archunit.postcompile;
+
+import com.tngtech.archunit.base.DescribedPredicate;
+import com.tngtech.archunit.core.domain.JavaClass;
+import com.tngtech.archunit.core.domain.JavaAccess;
+import com.tngtech.archunit.lang.ArchCondition;
+import com.tngtech.archunit.lang.ConditionEvent;
+import com.tngtech.archunit.lang.ConditionEvents;
+import com.tngtech.archunit.lang.SimpleConditionEvent;
+import com.tngtech.archunit.thirdparty.com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.HashSet;
+import java.util.Collections;
+
+import static com.tngtech.archunit.lang.ConditionEvent.createMessage;
+import static com.tngtech.archunit.thirdparty.com.google.common.base.Preconditions.checkNotNull;
+import static com.tngtech.archunit.thirdparty.com.google.common.collect.Iterables.getLast;
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toSet;
+
+/**
+ * Checks that a class transitively accesses methods that match a given predicate.
+ */
+public class TransitivelyAccessesMethodsCondition extends ArchCondition<JavaClass> {
+    /**
+     * Predicate to match the accessed methods
+     */
+    private final DescribedPredicate<? super JavaAccess<?>> conditionPredicate;
+
+    /**
+     * Transitive access path to find the path to the accessed method
+     */
+    private final TransitiveAccessPath transitiveAccessPath = new TransitiveAccessPath();
+
+    /**
+     * Custom class resolver to resolve classes that are outside classpath to be able to analyze them transitively
+     */
+    private final CustomClassResolver customClassResolver;
+
+    /**
+     * @param conditionPredicate Predicate to match the accessed methods
+     */
+    public TransitivelyAccessesMethodsCondition(DescribedPredicate<? super JavaAccess<?>> conditionPredicate) {
+        super("transitively depend on classes that " + conditionPredicate.getDescription());
+        this.conditionPredicate = checkNotNull(conditionPredicate);
+        this.customClassResolver = new CustomClassResolver();
+    }
+
+    /**
+     * Checks that a class transitively accesses methods that match a given predicate.
```

Comment:
i think these comments can be removed as the code is readable here

Reply:
This documents a class.